sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala

```diff
@@ -212,25 +212,30 @@ class SQLAppStatusListener(
     val maxMetricsFromAllStages = new mutable.HashMap[Long, Array[Long]]()
 
     taskMetrics.foreach { case (id, values) =>
```

Comment:
to reduce code diff, how about `taskMetrics.filter(m => metricTypes.contains(m.id)).foreach...`